homeassistant/components/fitbit/sensor.py

```diff
@@ -397,271 +443,153 @@ class FitbitSensorEntityDescription(SensorEntityDescription):
-            api.async_get_unit_system(), hass.loop
-        ).result()
-
-        clock_format = config[CONF_CLOCK_FORMAT]
-        monitored_resources = config[CONF_MONITORED_RESOURCES]
-        resource_list = [
-            *FITBIT_RESOURCES_LIST,
-            SLEEP_START_TIME_12HR if clock_format == "12H" else SLEEP_START_TIME,
-        ]
-        entities = [
-            FitbitSensor(
-                api,
-                user_profile.encoded_id,
-                config_path,
-                description,
-                units=description.unit_fn(unit_system),
-            )
-            for description in resource_list
-            if description.key in monitored_resources
-        ]
-        if "devices/battery" in monitored_resources:
-            devices = asyncio.run_coroutine_threadsafe(
-                api.async_get_devices(),
-                hass.loop,
-            ).result()
-            entities.extend(
-                [
-                    FitbitSensor(
-                        api,
-                        user_profile.encoded_id,
-                        config_path,
-                        FITBIT_RESOURCE_BATTERY,
-                        device,
-                    )
-                    for device in devices
-                ]
-            )
-        add_entities(entities, True)
-
-    else:
-        oauth = FitbitOauth2Client(
-            config_file.get(CONF_CLIENT_ID), config_file.get(CONF_CLIENT_SECRET)
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_IMPORT},
+                data={
+                    "auth_implementation": DOMAIN,
+                    CONF_TOKEN: {
+                        ATTR_ACCESS_TOKEN: config_file[ATTR_ACCESS_TOKEN],
+                        ATTR_REFRESH_TOKEN: config_file[ATTR_REFRESH_TOKEN],
+                        "expires_at": config_file[ATTR_LAST_SAVED_AT],
+                    },
+                    CONF_CLOCK_FORMAT: config[CONF_CLOCK_FORMAT],
+                    CONF_UNIT_SYSTEM: config[CONF_UNIT_SYSTEM],
+                    CONF_MONITORED_RESOURCES: config[CONF_MONITORED_RESOURCES],
+                },
+            ),
         )
+        issue_key = "deprecated_yaml_import"
+        translation_key = "deprecated_yaml_import"
+    else:
+        issue_key = "deprecated_yaml_no_import"
+        translation_key = "deprecated_yaml_no_import"
 
-        redirect_uri = f"{get_url(hass, require_ssl=True)}{FITBIT_AUTH_CALLBACK_PATH}"
-
-        fitbit_auth_start_url, _ = oauth.authorize_token_url(
-            redirect_uri=redirect_uri,
-            scope=[
-                "activity",
-                "heartrate",
-                "nutrition",
-                "profile",
-                "settings",
-                "sleep",
-                "weight",
-            ],
-        )
+    try:
+        await hass.async_add_executor_job(os.unlink, config_path)
```

Comment:
If we remove the old config file the user can't try again if the import fails. I'd not remove the file, and let the user do that.